### DISCUSSION TITLE
[RFC] Enforce dot-notation in sub-routers

### DISCUSSION THREAD

**Discussion by KATT** [trpc/trpc#275] [Ideas]
Initially when writing trpc>1 I was thinking I would use `/` as a separator between sub-routers, but it's more readable to have the whole path written out.

I've forgotten to add the `.` at the end a few times when creating sub-routers and it's a bit annoying. Don't see any harm in enforcing it.

This would be a major version change, but easy refactor - could be done application-wide with a simple regex.

# Change proposal 
## Before

```tsx
const appRouter = createRouter()
  .merge('users.', users) // prefix user procedures with "users."
  .merge('posts.', posts) // prefix poosts procedures with "posts."
  ;
```

## After

```tsx
const appRouter = createRouter()
  .merge('users', users) // prefix user procedures with "users."
  .merge('posts', posts) // prefix poosts procedures with "posts."
  ;
```

**Comment by KATT** [comment]
cc @cyrus-za

**Comment by cyrus-za** [comment]
I think this can be achieved without making it a breaking change. Can have automatic alias for ones without it and then add a linting rule or something? But I am fine with making another major version. Which ever works for you.

I like the idea though. I've also ran into it

**Comment by KATT** [comment]
Sure, I could make a minor version with a deprecation warning!
